src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMemberAccessor.cs

```diff
@@ -87,87 +92,77 @@ internal sealed class ReflectionEmitMemberAccessor : MemberAccessor
                 generator.Emit(OpCodes.Ldarg_0);
                 generator.Emit(OpCodes.Ldc_I4_S, i);
                 generator.Emit(OpCodes.Ldelem_Ref);
-
-                if (paramType.IsValueType)
-                {
-                    generator.Emit(OpCodes.Unbox_Any, paramType);
-                }
-                else
-                {
-                    generator.Emit(OpCodes.Castclass, paramType);
-                };
+                generator.Emit(
+                    paramType.IsValueType
```

Comment:
I know this is pre-existing code, but since you're touching it - this doesn't need the IsValueType check. It can always be unbox_any:

![image](https://user-images.githubusercontent.com/13110571/82363269-dced1a00-9a0d-11ea-8b6a-303fcb6a7f4a.png)

Reply:
Good point!